active_3d_planning_voxblox/include/active_3d_planning_voxblox/map/uncertainty_aware_map.h

```diff
@@ -0,0 +1,73 @@
+#ifndef ACTIVE_3D_PLANNING_APP_CURIOSITY_UNCERTAINTY_AWARE_MAP_H
+#define ACTIVE_3D_PLANNING_APP_CURIOSITY_UNCERTAINTY_AWARE_MAP_H
+
+#include <memory>
+
+#include <active_3d_planning_core/module/module_factory_registry.h>
+#include <voxblox_ros/esdf_server.h>
+
+#include "active_3d_planning_core/map/voxel_value_map.h"
+
+#include "voxblox.h"
+
+namespace active_3d_planning {
+namespace map {
+
+// Map that consists of two voxblox maps. One is used for reconstruction whereas
+// the other is used to store an uncertainty value for each voxel
+class UncertaintyAwareMap : public VoxelValueMap {
+ public:
+  explicit UncertaintyAwareMap(PlannerI& planner);  // NOLINT
+
+  // implement virtual methods
+  void setupFromParamMap(Module::ParamMap* param_map) override;
+
+  // check collision for a single pose
+  bool isTraversable(const Eigen::Vector3d& position,
+                     const Eigen::Quaterniond& orientation) override;
+
+  // check whether point is part of the map
+  bool isObserved(const Eigen::Vector3d& point) override;
```

Comment:
The vast majority of these functions will be identical to the parent implementation, then you don't need to override them.